nssp/delphi_nssp/pull.py

```diff
@@ -47,18 +52,21 @@ def pull_nssp_data(socrata_token: str):
     pd.DataFrame
         Dataframe as described above.
     """
-    # Pull data from Socrata API
-    client = Socrata("data.cdc.gov", socrata_token)
-    results = []
-    offset = 0
-    limit = 50000  # maximum limit allowed by SODA 2.0
-    while True:
-        page = client.get("rdmq-nq56", limit=limit, offset=offset)
-        if not page:
-            break  # exit the loop if no more results
-        results.extend(page)
-        offset += limit
-    df_ervisits = pd.DataFrame.from_records(results)
+    if not issue_date:
+        # Pull data from Socrata API
+        client = Socrata("data.cdc.gov", socrata_token)
+        results = []
+        offset = 0
+        limit = 50000  # maximum limit allowed by SODA 2.0
+        while True:
+            page = client.get("rdmq-nq56", limit=limit, offset=offset)
+            if not page:
+                break  # exit the loop if no more results
+            results.extend(page)
+            offset += limit
+        df_ervisits = pd.DataFrame.from_records(results)
+    else:
+        df_ervisits = pd.read_csv(f"{source_dir}/{issue_date}.csv")
```

Comment:
What if this file doesn't exist? what should be the behavior while patching if the csv doesn't exist?

Reply:
For the `issue_date.csv`, I wrote [this](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-8fce7ae14a0e2097dd3da98817cd6258af5a7596b2a7117339da5ce8abffc875R118-R121) so no next steps (including `pull.py`) will be called with csv that don't exists because the script skip the date right at the start.
Similarly, I just added guardrails in `patch.py` for the source dir [here](https://github.com/cmu-delphi/covidcast-indicators/pull/2000/files#diff-8fce7ae14a0e2097dd3da98817cd6258af5a7596b2a7117339da5ce8abffc875R73-R75).